pandas/io/formats/style.py

```diff
@@ -4081,9 +4081,43 @@ def css_calc(x, left: float, right: float, align: str, color: str | list | tuple
         else:
             return ret
 
+    def replace_pd_NA_with_np_nan(data_structure):
+        """
+        Recursively replace pd.NA with np.nan in a nested list or array.
+
+        This function traverses a nested structure (like lists or numpy arrays)
+        and replaces occurrences of Pandas' NA values (pd.NA) with NumPy's NaN values (np.nan).
+        It handles nested lists and arrays, applying the replacement recursively.
+
+        Parameters:
+        data_structure (list, np.ndarray, or any): A nested list, numpy array, or any other object.
+                                                   If it's a list or numpy array, the function will
+                                                   process its elements recursively.
+
+        Returns:
+        list, np.ndarray, or any: A new object with the same structure as `data_structure`,
+                                  where all pd.NA values have been replaced with np.nan.
+                                  The type of the returned object is the same as the input.
+
+        Example:
+        >>> data = [1, pd.NA, [3, pd.NA, [pd.NA, 5]], pd.NA]
+        >>> replace_pd_NA_with_np_nan(data)
+        [1, nan, [3, nan, [nan, 5]], nan]
+        """
+        if isinstance(data_structure, list):
+            # Process each item in the list recursively
+            return [replace_pd_NA_with_np_nan(element) for element in data_structure]
+        elif isinstance(data_structure, np.ndarray):
+            # Convert numpy array elements recursively
+            return np.array([replace_pd_NA_with_np_nan(element) for element in data_structure])
```

Comment:
This looks quite non-performant for big sparse arrays.
Are there other approaches that might be better for this whole process.

Maybe there is a substitute for `np.nanmin` instead of wrangling the data to make it fit into the numpy function?

Reply:
Hi @attack68 , thanks for your advice! I have implemented an alternative - using `df.min/max` followed by `np.nanmin/nanmax`. This will resolve the issues caused by numpy cannot properly handle pd.NA in certain circumstances. Here is the explanation for this issue: when import missing value with `pyarrow`, pd.NA will be generated in DataFrame, then, doing `np.nanmin(df.to_numpy())` in `_bar()`, numpy will throw a TypeError. However, after adding `df.min/max`, `pd.NA` will be properly handled and the returned structure will have no `pd.NA`, so it will be safe to proceed with `np.nanmin/nanmax` afterwords. This solution will resolve the issue without causing performance issue, although a little bit tricky.

Reply:
The alternative proposed will likely be much more performant, as well as being much more condensed and easy to read code